github_actions/lib/dependabot/github_actions/file_fetcher.rb

```diff
@@ -40,7 +40,8 @@ def workflow_files
         @workflow_files ||=
           repo_contents(dir: ".github/workflows", raise_errors: false).
           select { |f| f.type == "file" && f.name.match?(/\.ya?ml$/) }.
-          map { |f| fetch_file_from_host(".github/workflows/#{f.name}") }
+          map { |f| fetch_file_from_host(".github/workflows/#{f.name}") } \
+          + [fetch_file_if_present("action.yml"), fetch_file_if_present("action.yaml")].compact
```

Comment:
Why do we use `fetch_file_from_host` for the workflow files but `fetch_file_if_present` for the action.yaml? I wonder if we should filter out the action.yaml from `repo_contents(dir: "/", raise_errors: false)` and then `fetch_file_from_host`, or if that would just needlessly complicate things. Not sure that I'm getting at a suggestion but the code stood out to me a little so maybe you can share a bit of context that I'm missing.

Reply:
No you are on the right track, I had the same thought initially. 
But in the top case we know that there exists a folder named `.github/workflows", and we should iterate through all the files in that one to find all `.ya?ml` files so there is a good reason to go through the entire folder.

In this new case we are only interested in looking at one file (however, with two possible file endings) so I thought it would be unnecessary to iterate through the entire root directory to find it so I looked in the other file_fetcher examples and found these functions that were used when we were only interested in one file. 

I have no issue with changing it to keep it similar to the first one, but it will be quite verbose compared to this one.

Reply:
Sounds reasonable to me, thanks for elaborating 👍